src/platform/linux.rs

```diff
@@ -45,28 +52,94 @@ impl WV for InnerWebView {
             if let Some(js) = msg.get_value() {
                 if let Some(context) = msg.get_global_context() {
                     if let Some(js) = js.to_string(&context) {
-                        let v: RPC = serde_json::from_str(&js).unwrap();
-                        let mut hashmap = CALLBACKS.lock().unwrap();
-                        let (f, d) = hashmap.get_mut(&(window_id, v.method)).unwrap();
-                        let status = f(d, v.id, v.params);
-
-                        let js = match status {
-                            Ok(()) => {
-                                format!(
-                                    r#"window._rpc[{}].resolve("RPC call success"); window._rpc[{}] = undefined"#,
-                                    v.id, v.id
-                                )
+                        match serde_json::from_str::<FuncCall>(&js) {
+                            Ok(mut ev) => {
+                                // Use `isize` to conform with existing `Callback` API but should 
+                                // really be a `u64`. Note that RPC spec allows for non-numbers 
+                                // in the `id` field!
+                                let id: i32 = if let Some(value) = ev.payload.id.clone().take() {
```

Comment:
Let's make Callback use u64 then.

Reply:
I can do this temporarily but I think I have a better idea. Bear with me - let's completely remove *all the calback code* and let people do it in Javascript:

```javascript
function someRpcMethod() {
  return window.rpc.call(arguments.callee.name, Arguments.prototype.slice.call(0));
}
```

That mimics what the `Callback` API gives us over the RPC channel! I think this is much more elegant.

As a convenience we could offer:

```rust
fn set_rpc_aliases( names: Vec<&str> );
```

Which would define those Javascript functions automatically.

Reply:
That sounds pretty great!